### PR TITLE
[Bugfix] Fix AutoRound pipeline inference to use sequential pipeline

### PR DESCRIPTION
**Pull request by changwangss**
**Summary**
This PR fixes pipeline inference for AutoRoundModifier so it is treated as a calibration-required modifier and runs with the sequential pipeline.

**Problem**
After PR #2131 ([Bugfix] Improve pipeline inference), CalibrationPipeline._infer_pipeline uses an explicit modifier-type check to decide whether calibration data is required.
AutoRoundModifier was not included in that check, so recipes containing AutoRound were inferred as datafree in some cases.

As a result:

DataFreePipeline only triggers CALIBRATION_EPOCH_START/END
SEQUENTIAL_EPOCH_END is never emitted
AutoRoundModifier.on_event never calls apply_autoround

**Root Cause**
AutoRoundModifier was omitted from the calibration-required modifier list in:

src/llmcompressor/pipelines/registry.py

**Fix**
Import AutoRoundModifier in pipeline registry
Add AutoRoundModifier to _modifier_requires_calibration(...) isinstance(...) tuple
This ensures AutoRound recipes are inferred as sequential, so SEQUENTIAL_EPOCH_END is emitted and apply_autoround is executed.

**Files Changed**
src/llmcompressor/pipelines/registry.py
Validation
Manual validation with AutoRound recipe shows inference changes from:

Inferred DataFreePipeline for AutoRoundModifier
to:
Inferred SequentialPipeline for AutoRoundModifier
and apply_autoround path becomes reachable.

**Notes**
This is a minimal fix and does not change behavior for other modifiers.

### CODE REVIEW

**Comment by changwangss** [comment]
@kylesayrs could you please take a quick look when you have a moment? This is a small follow-up fix related to the pipeline inference change from #2131. We found that `AutoRoundModifier` was omitted from the calibration-required modifier list, which causes AutoRound recipes to be inferred as `datafree` instead of `sequential`.

**Comment by changwangss** [comment]
@yiliu30, this PR fixes some AutoRound examples that require calibration. Could you please review it?
https://github.com/vllm-project/llm-compressor/tree/main/examples/autoround
https://github.com/vllm-project/llm-compressor/blob/main/examples/quantization_w8a8_mxfp8/autoround_qwen3_example.py

**Review by yiliu30 — APPROVED** [review]
Thanks for the fix, LGTM!

File: src/llmcompressor/pipelines/registry.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from torch.utils.data.dataloader import DataLoader
 
 from llmcompressor.modifiers import Modifier
+from llmcompressor.modifiers.autoround import AutoRoundModifier
```
[the remark this answers is not in the document]
Comment (yiliu30) [reply src/llmcompressor/pipelines/registry.py]: No worries, auto-round is the default dependency.

File: src/llmcompressor/pipelines/registry.py [hunk 2]
```diff
@@ -66,6 +67,7 @@ def _modifier_requires_calibration(modifier: Modifier):
                     SparsityModifierBase,
                     GPTQModifier,
                     AWQModifier,
+                    AutoRoundModifier,
                     IMatrixGatherer,
                 ),
             ):
```
[the remark this answers is not in the document]
Comment (changwangss) [reply src/llmcompressor/pipelines/registry.py]: Thanks for the suggestion. In this repository, `auto-round` is already a default dependency, so this import does not introduce a new optional dependency edge.

I kept the fix minimal here since the issue is that `AutoRoundModifier` is currently omitted from pipeline inference and therefore does not trigger the sequential pipeline.

**Comment by changwangss** [comment]
@brian-dellabetta
@shanjiaz
@HDCharles
Could a maintainer please add the `ready` label? The PR is code complete and I have validated the fix locally.

File: src/llmcompressor/pipelines/registry.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (brian-dellabetta) [reply src/llmcompressor/pipelines/registry.py]: It's not a big issue, but we set up a bad pattern here that gets more important for AutoRoundModifier because it requires `import autoround` at top level, even though a user might be running something that has nothing to do with it. Mind removing the imports and relying on the string match instead? See comment below

File: src/llmcompressor/pipelines/registry.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (brian-dellabetta) [reply src/llmcompressor/pipelines/registry.py]: yeah we can do 
```python
if modifier.__class__.__name__ in ("SmoothQuantModifier", "WandaPruningModifier", "SparseGPTModifier", "GPTQModifier", "AWQModifier", "AutoRoundModifier", "IMatrixGatherer")
```

**Review by brian-dellabetta — COMMENTED** [review]
Thanks @changwangss for catching this! Gemini's comments are minor, but probably worth addressing. Mind making the changes suggested below? Please also add a test case to `tests/llmcompressor/pipelines/test_registry.py:test_infer_pipeline`. Something like 
```python
([AutoRoundModifier()], SequentialPipeline),
```

File: src/llmcompressor/pipelines/registry.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (HDCharles) [reply src/llmcompressor/pipelines/registry.py]: Agree avoiding the import is better

File: src/llmcompressor/pipelines/registry.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yiliu30) [reply src/llmcompressor/pipelines/registry.py]: Sure, make sense.

**Comment by changwangss** [comment]
Updated per review: removed the top-level AutoRound import and switched to class-name based detection for `AutoRoundModifier` to avoid introducing a hard dependency edge in the pipeline registry.
@brian-dellabetta @HDCharles please review again.

**Review by brian-dellabetta — APPROVED** [review]
i added a check for AutoRoundModifier to our infer pipeline tests on this PR as well. Thanks for catching and for updating!

File: src/llmcompressor/pipelines/registry.py [hunk 3]
```diff
@@ -59,15 +56,14 @@ def from_modifiers(
...
-            if isinstance(
-                modifier,
-                (
-                    SmoothQuantModifier,
-                    SparsityModifierBase,
-                    GPTQModifier,
-                    AWQModifier,
-                    IMatrixGatherer,
-                ),
+            if modifier.__class__.__name__ in (
```
Comment (dsikka) [inline src/llmcompressor/pipelines/registry.py:59]: Why not just import and check as was done previously?

Comment (brian-dellabetta) [reply src/llmcompressor/pipelines/registry.py:59]: Yeah the original PR imported the class directly and used an isinstance check, following the previous pattern, but we asked them to update on this PR to avoid potentially unnecessary imports at a higher level scope. The updates to unit tests though means that we're testing this

**Review by HDCharles — APPROVED** [review]

**Review by dsikka — APPROVED** [review]
